tf2/include/tf2/transform_datatypes.h

```diff
@@ -88,6 +88,63 @@ bool operator==(const Stamped<T> & a, const Stamped<T> & b)
          static_cast<const T &>(a) == static_cast<const T &>(b);
 }
 
+/** \brief The data type which will be cross compatable with geometry_msgs
+ * This is the tf2 datatype equivalent of a MessageWithCovarianceStamped */
+template<typename T>
+class WithCovarianceStamped : public T
+{
+public:
+  TimePoint stamp_;   ///< The timestamp associated with this data
+  std::string frame_id_;   ///< The frame_id associated this data
+  std::array<std::array<double, 6>, 6> cov_mat_;  ///< The covariance matrix associated with this data
+
+  /** Default constructor */
+  WithCovarianceStamped()
+  : frame_id_("NO_ID_STAMPED_DEFAULT_CONSTRUCTION"),
+    cov_mat_{}
+  {
+  }
+
+  /** Full constructor */
+  WithCovarianceStamped(
+    const T & input,
```

Comment:
It doesn't look like we use this input at all.  Do we need it?

Reply:
This was actually a bug - I didn't fully fill out the copy constructor. Fixed in 34246f8.